src/modules/user/user.service.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  Inject,
+  Injectable,
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { UserRepository } from './user.repository';
+import { User } from '@prisma/client';
+
+@Injectable()
+export class UserService {
+  constructor(
+    @Inject('UserRepository') private readonly userRepository: UserRepository,
+  ) {}
+
+  async login(email: string, password: string): Promise<User | null> {
+    const user = await this.userRepository.findByEmail(email);
+
+    if (!user) {
+      throw new NotFoundException(`User with username ${email} not found`);
+    }
+    const isPasswordValid = user.password === password;
+
+    if (!isPasswordValid) {
```

Comment:
It's never a good idea to throw credential specific errors, this opens up applications to brute force attacks and the attacker now knows that there is a user with a given email for which the password is missing.